master/reference/cni-plugin/configuration.md

```diff
@@ -180,3 +200,70 @@ When using the CNI `host-local` IPAM plugin, a special value `usePodCidr` is all
 ```
 
 When making use of the `usePodCidr` option, the Calico CNI plugin requires read-only Kubernetes API access to the `Nodes` resource.
+
+##### IPAM Manipulation with Kubernetes Annotations
+
+###### Specifying IP Pools on a per-Pod basis
+
+In addition to specifying IP Pools in the CNI config as discussed above, Calico IPAM supports specifying IP Pools per-Pod using the following [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/).
+
+- `cni.projectcalico.org/ipv4pools`: A list of configured IPv4 Pools from which to choose an address for the Pod.
+
+   Example:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv4pools": "[\"192.168.0.0/16\"]"
+```
+
+- `cni.projectcalico.org/ipv6pools`: A list of configured IPv6 Pools from which to choose an address for the Pod.
+
+   Example:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv6pools": "[\"2001:db8::1/120\"]"
+```
+
+If provided, these IP Pools will override any IP Pools specified in the CNI config.
+
+
+
+  > **Note:**
+  >
+  > This requires the IP Pools to exist before `ipv4pools` or `ipv6pools` annotations are used.
+  > Requesting a subset of an IP Pool is not supported. IP Pools requested in the annotations must be the exact match to the IP Pools created.
+
+###### Requesting a Specific IP address
+
+You can also request a specific IP address through [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/) with Calico IPAM. 
+There are two annotations to request a specific IP address:
+
+- `cni.projectcalico.org/ipAddrs`: A list of IPv4 and/or IPv6 addresses to assign to the Pod. The requested IP addresses will be assigned from Calico IPAM and must exist within a configured IP Pool.
+
+  Example:
+
+```yaml
+annotations:
+        "cni.projectcalico.org/ipAddrs": "[\"192.168.0.1\"]"
+```
+
+- `cni.projectcalico.org/ipAddrsNoIpam`: A list of IPv4 and/or IPv6 addresses to assign to the Pod, bypassing IPAM. Any IP conflicts and routing have to be taken care of manually or by some other system.
```

Comment:
I think this could use another sentence about route distribution if the IP is in a Calico pool

e.g "Calico will only distribute routes to a Pod if its IP address falls within a Calico IP Pool.  If you assign an IP address that is not in a Calico IP Pool, you must ensure that routing to that IP address is taken care of through another mechanism."